File: suc/swift/095.swift
import SwiftUI

@main
struct ReceiptApp: App {
    var body: some Scene {
        WindowGroup {
            ContentView()
        }
    }
}

struct ContentView: View {
    var body: some View {
        ReceiptScreen()
            .preferredColorScheme(.light)
            .statusBarHidden(true)
    }
}

struct ReceiptScreen: View {
    var body: some View {
        ZStack {
            Color.white.ignoresSafeArea()
            
            VStack(alignment: .center, spacing: 0) {
                // Header
                Spacer().frame(height: 20)
                
                SuccessIcon()
                
                Spacer().frame(height: 16)
                
                Text("Payment Success")
                    .font(.system(size: 16, weight: .medium)) // TitleMedium approx
                    .foregroundColor(.gray)
                
                Spacer().frame(height: 8)
                
                Text("$1,249.00")
                    .font(.system(size: 45, weight: .bold)) // DisplayMedium approx
                    .foregroundColor(.black)
                
                Spacer().frame(height: 32)
                
                // Receipt Card
                ReceiptDetailsCard()
                
                Spacer()
                
                // Action Buttons
                ActionButtons()
                
                Spacer().frame(height: 16)
            }
            .padding(24)
        }
    }
}

struct SuccessIcon: View {
    var body: some View {
        ZStack {
            Circle()
                .fill(Color(red: 0.91, green: 0.96, blue: 0.91)) // #E8F5E9 (Light Green)
                .frame(width: 72, height: 72)
            
            Image(systemName: "checkmark")
                .resizable()
                .aspectRatio(contentMode: .fit)
                .frame(width: 36, height: 36)
                .foregroundColor(Color(red: 0.18, green: 0.49, blue: 0.20)) // #2E7D32 (Darker Green)
                .font(Font.system(size: 36, weight: .bold)) // Make checkmark thicker
        }
    }
}

struct ReceiptDetailsCard: View {
    var body: some View {
        VStack(spacing: 16) {
            // Recipient
            RowDetail(
                label: "To Merchant",
                value: "TechSpace Solutions",
                iconName: "info.circle.fill"
            )
            
            Divider()
                .background(Color(red: 0.88, green: 0.88, blue: 0.88)) // #E0E0E0
            
            // Date
            RowDetail(
                label: "Date",
                value: "Dec 02, 2025 - 11:30 PM",
                iconName: "calendar"
            )
            
            Divider()
                .background(Color(red: 0.88, green: 0.88, blue: 0.88)) // #E0E0E0
            
            // Transaction ID
            HStack {
                Text("Ref Number")
                    .font(.system(size: 14)) // BodyMedium approx
                    .foregroundColor(.gray)
                
                Spacer()
                
                Text("009281-XX-99")
                    .font(.system(size: 14, weight: .medium))
                    .foregroundColor(.black)
            }
            
            // Payment Method
            HStack {
                Text("Payment Method")
                    .font(.system(size: 14))
                    .foregroundColor(.gray)
                
                Spacer()
                
                Text("Credit Card •••• 4242")
                    .font(.system(size: 14, weight: .medium))
                    .foregroundColor(.black)
            }
        }
        .padding(20)
        .background(Color(red: 0.97, green: 0.98, blue: 0.98)) // #F8F9FA
        .cornerRadius(12)
        .overlay(
            RoundedRectangle(cornerRadius: 12)
                .stroke(Color(red: 0.93, green: 0.93, blue: 0.93), lineWidth: 1) // #EEEEEE
        )
    }
}

struct RowDetail: View {
    let label: String
    let value: String
    let iconName: String
    
    var body: some View {
        HStack {
            HStack(spacing: 8) {
                Image(systemName: iconName)
                    .resizable()
                    .aspectRatio(contentMode: .fit)
                    .frame(width: 18, height: 18)
                    .foregroundColor(.gray)
                
                Text(label)
                    .font(.system(size: 14))
                    .foregroundColor(.gray)
            }
            
            Spacer()
            
            Text(value)
                .font(.system(size: 14, weight: .semibold))
                .foregroundColor(.black)
        }
    }
}

struct ActionButtons: View {
    var body: some View {
        VStack(spacing: 12) {
            Button(action: {}) {
                Text("Done")
                    .font(.system(size: 16, weight: .bold)) // BodyLarge + Bold
                    .foregroundColor(.white)
                    .frame(maxWidth: .infinity)
                    .frame(height: 50)
                    .background(Color.black)
                    .cornerRadius(8)
            }
            
            Button(action: {}) {
                HStack(spacing: 8) {
                    Image(systemName: "square.and.arrow.up") // Share icon approximation
                        .resizable()
                        .aspectRatio(contentMode: .fit)
                        .frame(width: 18, height: 18)
                        .foregroundColor(.black)
                    
                    Text("Share Receipt")
                        .font(.system(size: 16)) // BodyLarge
                        .foregroundColor(.black)
                }
                .frame(maxWidth: .infinity)
                .frame(height: 50)
                .overlay(
                    RoundedRectangle(cornerRadius: 8)
                        .stroke(Color(red: 0.87, green: 0.87, blue: 0.87), lineWidth: 1) // #DDDDDD
                )
            }
        }
    }
}

File: suc/kotlin/095.kt
package com.example.myapplication

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.activity.enableEdgeToEdge
import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Check
import androidx.compose.material.icons.filled.DateRange
import androidx.compose.material.icons.filled.Info
import androidx.compose.material.icons.filled.Share
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.HorizontalDivider
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.OutlinedButton
import androidx.compose.material3.Scaffold
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.SideEffect
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.toArgb
import androidx.compose.ui.platform.LocalView
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.core.view.WindowCompat
import androidx.core.view.WindowInsetsCompat
import androidx.core.view.WindowInsetsControllerCompat

class MainActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContent {
            // Apply a local theme setup
            MaterialTheme(
                colorScheme = MaterialTheme.colorScheme.copy(
                    background = Color.White,
                    surface = Color.White,
                    onBackground = Color.Black,
                    onSurface = Color.Black
                )
            ) {
                // Handle Immersive Mode (Hide System Bars)
                ImmersiveSystemBars()

                ReceiptScreen()
            }
        }
    }
}

@Composable
fun ImmersiveSystemBars() {
    val view = LocalView.current
    if (!view.isInEditMode) {
        SideEffect {
            val window = (view.context as ComponentActivity).window
            val insetsController = WindowCompat.getInsetsController(window, view)

            // Hide both status bar and navigation bar
            insetsController.hide(WindowInsetsCompat.Type.systemBars())

            // Allow swiping to show bars temporarily
            insetsController.systemBarsBehavior =
                WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE

            // Ensure status bar color is transparent to avoid conflicts
            window.statusBarColor = Color.Transparent.toArgb()
            window.navigationBarColor = Color.Transparent.toArgb()
        }
    }
}

@Composable
fun ReceiptScreen() {
    Scaffold(
        containerColor = Color.White,
        modifier = Modifier.fillMaxSize()
    ) { innerPadding ->
        Column(
            modifier = Modifier
                .padding(innerPadding)
                .fillMaxSize()
                .padding(24.dp),
            horizontalAlignment = Alignment.CenterHorizontally
        ) {

            // --- Header Section ---
            Spacer(modifier = Modifier.height(20.dp))
            SuccessIcon()
            Spacer(modifier = Modifier.height(16.dp))
            Text(
                text = "Payment Success",
                style = MaterialTheme.typography.titleMedium,
                color = Color.Gray
            )
            Spacer(modifier = Modifier.height(8.dp))
            Text(
                text = "$1,249.00",
                style = MaterialTheme.typography.displayMedium,
                fontWeight = FontWeight.Bold,
                color = Color.Black
            )
            Spacer(modifier = Modifier.height(32.dp))

            // --- Receipt Card ---
            ReceiptDetailsCard()

            Spacer(modifier = Modifier.weight(1f))

            // --- Action Buttons ---
            ActionButtons()
            Spacer(modifier = Modifier.height(16.dp))
        }
    }
}

@Composable
fun SuccessIcon() {
    Box(
        modifier = Modifier
            .size(72.dp)
            .background(Color(0xFFE8F5E9), CircleShape), // Very light green
        contentAlignment = Alignment.Center
    ) {
        Icon(
            imageVector = Icons.Default.Check,
            contentDescription = "Success",
            tint = Color(0xFF2E7D32), // Darker Green
            modifier = Modifier.size(36.dp)
        )
    }
}

@Composable
fun ReceiptDetailsCard() {
    Card(
        modifier = Modifier.fillMaxWidth(),
        shape = RoundedCornerShape(12.dp),
        colors = CardDefaults.cardColors(
            containerColor = Color(0xFFF8F9FA) // Very light gray for card background
        ),
        border = BorderStroke(1.dp, Color(0xFFEEEEEE))
    ) {
        Column(
            modifier = Modifier.padding(20.dp),
            verticalArrangement = Arrangement.spacedBy(16.dp)
        ) {
            // Recipient
            RowDetail(
                label = "To Merchant",
                value = "TechSpace Solutions",
                icon = Icons.Default.Info
            )

            HorizontalDivider(color = Color(0xFFE0E0E0))

            // Date
            RowDetail(
                label = "Date",
                value = "Dec 02, 2025 - 11:30 PM",
                icon = Icons.Default.DateRange
            )

            HorizontalDivider(color = Color(0xFFE0E0E0))

            // Transaction ID
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.CenterVertically
            ) {
                Text(
                    text = "Ref Number",
                    style = MaterialTheme.typography.bodyMedium,
                    color = Color.Gray
                )
                Text(
                    text = "009281-XX-99",
                    style = MaterialTheme.typography.bodyMedium,
                    fontWeight = FontWeight.Medium,
                    color = Color.Black
                )
            }

            // Payment Method
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.CenterVertically
            ) {
                Text(
                    text = "Payment Method",
                    style = MaterialTheme.typography.bodyMedium,
                    color = Color.Gray
                )
                Text(
                    text = "Credit Card •••• 4242",
                    style = MaterialTheme.typography.bodyMedium,
                    fontWeight = FontWeight.Medium,
                    color = Color.Black
                )
            }
        }
    }
}

@Composable
fun RowDetail(label: String, value: String, icon: androidx.compose.ui.graphics.vector.ImageVector) {
    Row(
        modifier = Modifier.fillMaxWidth(),
        horizontalArrangement = Arrangement.SpaceBetween,
        verticalAlignment = Alignment.CenterVertically
    ) {
        Row(verticalAlignment = Alignment.CenterVertically) {
            Icon(
                imageVector = icon,
                contentDescription = null,
                tint = Color.Gray,
                modifier = Modifier.size(18.dp)
            )
            Spacer(modifier = Modifier.width(8.dp))
            Text(
                text = label,
                style = MaterialTheme.typography.bodyMedium,
                color = Color.Gray
            )
        }
        Text(
            text = value,
            style = MaterialTheme.typography.bodyMedium,
            fontWeight = FontWeight.SemiBold,
            color = Color.Black
        )
    }
}

@Composable
fun ActionButtons() {
    Column(
        verticalArrangement = Arrangement.spacedBy(12.dp)
    ) {
        Button(
            onClick = { /* No logic allowed */ },
            modifier = Modifier
                .fillMaxWidth()
                .height(50.dp),
            colors = ButtonDefaults.buttonColors(
                containerColor = Color.Black,
                contentColor = Color.White
            ),
            shape = RoundedCornerShape(8.dp)
        ) {
            Text(
                text = "Done",
                style = MaterialTheme.typography.bodyLarge,
                fontWeight = FontWeight.Bold
            )
        }

        OutlinedButton(
            onClick = { /* No logic allowed */ },
            modifier = Modifier
                .fillMaxWidth()
                .height(50.dp),
            shape = RoundedCornerShape(8.dp),
            border = BorderStroke(1.dp, Color(0xFFDDDDDD))
        ) {
            Icon(
                imageVector = Icons.Default.Share,
                contentDescription = null,
                modifier = Modifier.size(18.dp),
                tint = Color.Black
            )
            Spacer(modifier = Modifier.width(8.dp))
            Text(
                text = "Share Receipt",
                style = MaterialTheme.typography.bodyLarge,
                color = Color.Black
            )
        }
    }
}

@Preview(showBackground = true)
@Composable
fun PreviewReceiptScreen() {
    MaterialTheme(
        colorScheme = MaterialTheme.colorScheme.copy(
            background = Color.White,
            surface = Color.White
        )
    ) {
        ReceiptScreen()
    }
}